activesupport/lib/active_support/encrypted_file.rb

```diff
@@ -73,7 +73,9 @@ def writing(contents)
         tmp_path = Pathname.new File.join(Dir.tmpdir, tmp_file)
         tmp_path.binwrite contents
 
-        yield tmp_path
+        # Windows paths can contained shortened forms, for example MyLongName can become MyLo~1
+        # This translate that shortened path into the absolute path
+        yield Pathname.new(Dir.glob(tmp_path)[0])
```

Comment:
I think this would break paths that have wildcards in their names:

```ruby
FileUtils.touch("a")
the_path_to_madness = "a*"
FileUtils.touch(the_path_to_madness)
Dir.glob(the_path_to_madness)[0]
# => "a"
```

Perhaps we could `glob` only when we detect an 8.3 format name?

It would be nice if https://bugs.ruby-lang.org/issues/12656 (or similar) were merged.  Then we could use `File.expand_path` instead.  Or, actually, it looks like `Dir.tmpdir` would [take care of itself](https://github.com/ruby/tmpdir/blob/c4802265c8f95698e933a43eae6dea7c30216b5f/lib/tmpdir.rb#L25).

Perhaps another option would be to avoid `Dir.tmpdir`, and use the app `tmp` directory instead?  Or would that suffer from the same 8.3 issue on Windows?

Reply:
After playing around with a few things, I opted to calling `Dir.glob` on `Dir.tmpdir` only. This is a "static" value, so there would be no wildcards. We're manually joining the path to the filename later on, so windows won't get in the way and shorten anything after this. 

The temp directory must be coming from windows somehow, and that must be where the shortened path is coming from, so that's the only thing we have to account for.

Reply:
`Dir.tmpdir` can be [read from one of a few different environment variables](https://github.com/ruby/tmpdir/blob/c4802265c8f95698e933a43eae6dea7c30216b5f/lib/tmpdir.rb#L23-L24).  In other words, it can be user-specified, and so might contain wildcards (as inadvisable as that would be).

I'm curious -- if you run the following in a Windows command prompt:

```
echo %TMPDIR% 
echo %TMP% 
echo %TEMP%
```

Does the 8.3 formatted path show up?  If so, does setting the relevant environment variable to the expanded path fix the issue?